charts/consul/templates/_helpers.tpl

```diff
@@ -682,4 +683,22 @@ Usage: {{ template "consul.versionInfo" }}
     {{- $sanitizedVersion = $versionInfo }}
 {{- end -}}
 {{- printf "%s" $sanitizedVersion | trunc 63 | quote }}
+{{- end -}}
+
+{{/*
+Sets the imagePullPolicy for all Consul images (consul, consul-dataplane, consul-k8s, consul-telemetry-collector)
+Valid values are:
+    IfNotPresent
+    Always
+    Never
+    In the case of empty, see https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy for details
+
+Usage: {{ template "consul.imagePullPolicy" . }} TODO: melisa should we name this differently ?
+*/}}
+{{- define "consul.imagePullPolicy" -}}
+{{ if or (eq .Values.global.imagePullPolicy "IfNotPresent") (eq .Values.global.imagePullPolicy "Always") (eq .Values.global.imagePullPolicy "Never")}}imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+{{ else if eq .Values.global.imagePullPolicy "" }}
```

Comment:
do we want a default value for this?

Reply:
Yeah I was trying to decide, because kubernetes does its own thing with pull policies, and there isn't just one default (https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting )

Reply:
It does accept empty string though.